src/magicbox/SuggestionsManager.ts

```diff
@@ -306,29 +336,85 @@ export class SuggestionsManager {
     return $$(dom);
   }
 
-  private move(direction: 'up' | 'down') {
+  private move(direction: direction) {
+    const previewSelectables = $$(this.element).findAll(`.coveo-preview-selectable`);
+    if (previewSelectables.length > 0) {
+      return this.moveWithQuerySuggestPreview(direction);
+    }
+    return this.moveWithinSuggestion(direction);
+  }
+
+  private moveWithinSuggestion(direction: direction) {
     const currentlySelected = $$(this.element).find(`.${this.options.selectedClass}`);
     const selectables = $$(this.element).findAll(`.${this.options.selectableClass}`);
     const currentIndex = indexOf(selectables, currentlySelected);
 
     let index = direction == 'up' ? currentIndex - 1 : currentIndex + 1;
-    if (index < -1) {
-      index = selectables.length - 1;
-    }
-    if (index > selectables.length) {
-      index = 0;
-    }
+    index = (index + selectables.length) % selectables.length;
 
-    const newlySelected = selectables[index];
+    this.lastSelectedSuggestion = selectables[index];
 
-    if (newlySelected) {
-      this.processKeyboardSelection(newlySelected);
+    return this.selectQuerySuggest(this.lastSelectedSuggestion);
+  }
+
+  private selectQuerySuggest(suggestion: HTMLElement) {
+    if (suggestion) {
+      this.processKeyboardSelection(suggestion);
     } else {
       this.keyboardFocusedSuggestion = null;
       this.inputManager.input.removeAttribute('aria-activedescendant');
     }
 
-    return newlySelected;
+    return suggestion;
+  }
+
+  private moveWithQuerySuggestPreview(direction: direction) {
+    const currentlySelected = $$(this.element).find(`.${this.options.selectedClass}`);
+    const omniboxSelectables = $$(this.element).findAll(`.${this.options.selectableClass}`);
+    const previewSelectables = $$(this.element).findAll(`.coveo-preview-selectable`);
+    const omniboxIndex = indexOf(omniboxSelectables, currentlySelected);
+    const previewIndex = indexOf(previewSelectables, currentlySelected);
+    const noSelection = omniboxIndex == -1 && previewIndex == -1;
+
+    //if we move within the suggestion
```

Comment:
Rather than adding a comment, we could store the condition in a well-named variable which serves as inline documentation.

e.g.

```
const isVerticalMove = direction == 'up' || direction == 'down';
const movedWithinSuggestion = (noSelection || omniboxIndex > -1) && isVerticalMove
```

For the condition, I don't understand the first half:
`noSelection` says  `omniboxIndex` is -1, and then we have `omniboxIndex > -1`.

So it seems the `omniboxIndex` index can be any number?
But if it is `-1`, then the previewIndex should also be -1. Wouldn't that always be the case?

Reply:
You are right, I don't need to have the `noSelection`

I did this before I modified the `move` function so I needed it because I couldn't move when nothing was selected 

I will delete it